### PR TITLE
circulation: update fee circulation tab.

### PR DESCRIPTION
**Pull request by zannkukai**
Reorganise the patron circulation fee tab adding the fees preview about
the overdue loans. Use a vertical tabs design to keep good readability
of all fees categories into the component.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Dependencies

Use the "US-cipo" branch from rero-ils repository

## How to test?

- Visit the circulation fee tab for a patron with overdue loans. Overdue settings should be defined into Circulation policies.
image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by Garfield-fr — APPROVED** [review]

**Review by AoNoOokami — APPROVED** [review]

File: projects/admin/src/app/classes/loans.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/* tslint:disable */
+// required as json properties is not lowerCamelCase
+
+import { marker } from '@biesbjerg/ngx-translate-extract-marker';
```
Comment (lauren-d) [inline projects/admin/src/app/classes/loans.ts:20]: in other PR, I see `import { marker as _ } from '@biesbjerg/ngx-translate-extract-marker';` what is better?

File: projects/admin/src/app/classes/loans.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  document_pid?: string;
+  item_pid?: {type: string, value: string};
+  start_date?: Moment;
+  end_date?: Moment;
+  request_expire_date?: Moment;
+  pickup_location_pid?: string;
+  item_destination?: LoanDestination;
+  transaction_location_pid?: string;
+
+  constructor(obj?: any) {
```
Comment (lauren-d) [inline projects/admin/src/app/classes/loans.ts:74]: missing docstring

File: projects/admin/src/app/classes/loans.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    Object.assign(this, obj);
+    this.request_expire_date = Loan._convertToMoment(this.request_expire_date);
+    this.start_date = Loan._convertToMoment(this.start_date);
+    this.end_date = Loan._convertToMoment(this.end_date);
+    this.transaction_date = Loan._convertToMoment(this.transaction_date);
+  }
+
+  private static _convertToMoment(data) {
```
Comment (lauren-d) [inline projects/admin/src/app/classes/loans.ts:82]: missing docstring

File: projects/admin/src/app/classes/loans.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    return (data)
+      ? moment(data)
+      : null;
+  }
+
+  get dueDate(): moment {
```
Comment (lauren-d) [inline projects/admin/src/app/classes/loans.ts:88]: missing doctstring

File: projects/admin/src/app/classes/loans.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    switch (this.state) {
+      case LoanState.PENDING: return this.request_expire_date;
+      case LoanState.ITEM_ON_LOAN: return this.end_date;
+      default: return null;
+    }
+  }
+
+  public get expired() {
```
Comment (lauren-d) [inline projects/admin/src/app/classes/loans.ts:96]: missing doctring

File: projects/admin/src/app/classes/loans.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/classes/loans.ts:20]: I tried but it seems to have a problem too use this syntax into `enum`. This syntax is already used in `items.ts` file

**Review by lauren-d — APPROVED** [review]

**Pull request merged**